### ISSUE TITLE
curve slicing

### ISSUE THREAD

**Issue by dmarx** [dmarx/keyframed#36]
```python
c = Curve({0:0, 10:10}, default_interpolation='linear')
c2 = c[2:5]
c2.duration == 4
c2[1] ==0
c2[2] ==2
c2.keyframes == [2,4]
```

**Comment by dmarx** [comment]
hmm... should a curve slice reindex the keyframes such that the first slice index maps to keyframe 0 in the returned value of the slice operation?

actually... some kind of reindex operator or relative index adjustment might be more valuable than the slice stuff by itself. could fold it into the easing implementation (which is definitely way overcomplicated)

**Comment by dmarx** [comment]
closed by #65
